internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go

```diff
@@ -207,12 +207,14 @@ func resourceDataFactoryIntegrationRuntimeAzureSsis() *pluginsdk.Resource {
 						"pricing_tier": {
 							Type:     pluginsdk.TypeString,
 							Optional: true,
-							Default:  string(datafactory.IntegrationRuntimeSsisCatalogPricingTierBasic),
 							ValidateFunc: validation.StringInSlice([]string{
-								string(datafactory.IntegrationRuntimeSsisCatalogPricingTierBasic),
-								string(datafactory.IntegrationRuntimeSsisCatalogPricingTierStandard),
-								string(datafactory.IntegrationRuntimeSsisCatalogPricingTierPremium),
-								string(datafactory.IntegrationRuntimeSsisCatalogPricingTierPremiumRS),
+								"Basic",
+								"S0", "S1", "S2", "S3", "S4", "S6", "S7", "S9", "S12",
+								"P1", "P2", "P4", "P6", "P11", "P15",
+								"GP_S_Gen5_1", "GP_S_Gen5_2", "GP_S_Gen5_4", "GP_S_Gen5_6", "GP_S_Gen5_8", "GP_S_Gen5_10", "GP_S_Gen5_12", "GP_S_Gen5_14", "GP_S_Gen5_16", "GP_S_Gen5_18", "GP_S_Gen5_20", "GP_S_Gen5_24", "GP_S_Gen5_32", "GP_S_Gen5_40",
+								"GP_Gen5_2", "GP_Gen5_4", "GP_Gen5_6", "GP_Gen5_8", "GP_Gen5_10", "GP_Gen5_12", "GP_Gen5_14", "GP_Gen5_16", "GP_Gen5_18", "GP_Gen5_20", "GP_Gen5_24", "GP_Gen5_32", "GP_Gen5_40", "GP_Gen5_80",
+								"BC_Gen5_2", "BC_Gen5_4", "BC_Gen5_6", "BC_Gen5_8", "BC_Gen5_10", "BC_Gen5_12", "BC_Gen5_14", "BC_Gen5_16", "BC_Gen5_18", "BC_Gen5_20", "BC_Gen5_24", "BC_Gen5_32", "BC_Gen5_40", "BC_Gen5_80",
+								"HS_Gen5_2", "HS_Gen5_4", "HS_Gen5_6", "HS_Gen5_8", "HS_Gen5_10", "HS_Gen5_12", "HS_Gen5_14", "HS_Gen5_16", "HS_Gen5_18", "HS_Gen5_20", "HS_Gen5_24", "HS_Gen5_32", "HS_Gen5_40", "HS_Gen5_80",
```

Comment:
Given we take this string and convert it into the pricing enum, will these not fail? or do they exist in the SDK?

Reply:
In https://github.com/Azure/azure-sdk-for-go/blob/main/services/datafactory/mgmt/2018-06-01/datafactory/enums.go.  It looks like the IntegrationRuntimeSsisCatalogPricingTier variable will accept any string, but there is the constant list and the PossibleIntegrationRuntimeSsisCatalogPricingTierValues function which does not return these additional values.  I am still looking through the SDK to see if the SDK will accept these additional values without error.
```
// IntegrationRuntimeSsisCatalogPricingTier enumerates the values for integration runtime ssis catalog pricing
// tier.
type IntegrationRuntimeSsisCatalogPricingTier string

const (
	// IntegrationRuntimeSsisCatalogPricingTierBasic ...
	IntegrationRuntimeSsisCatalogPricingTierBasic IntegrationRuntimeSsisCatalogPricingTier = "Basic"
	// IntegrationRuntimeSsisCatalogPricingTierPremium ...
	IntegrationRuntimeSsisCatalogPricingTierPremium IntegrationRuntimeSsisCatalogPricingTier = "Premium"
	// IntegrationRuntimeSsisCatalogPricingTierPremiumRS ...
	IntegrationRuntimeSsisCatalogPricingTierPremiumRS IntegrationRuntimeSsisCatalogPricingTier = "PremiumRS"
	// IntegrationRuntimeSsisCatalogPricingTierStandard ...
	IntegrationRuntimeSsisCatalogPricingTierStandard IntegrationRuntimeSsisCatalogPricingTier = "Standard"
)

// PossibleIntegrationRuntimeSsisCatalogPricingTierValues returns an array of possible values for the IntegrationRuntimeSsisCatalogPricingTier const type.
func PossibleIntegrationRuntimeSsisCatalogPricingTierValues() []IntegrationRuntimeSsisCatalogPricingTier {
	return []IntegrationRuntimeSsisCatalogPricingTier{IntegrationRuntimeSsisCatalogPricingTierBasic, IntegrationRuntimeSsisCatalogPricingTierPremium, IntegrationRuntimeSsisCatalogPricingTierPremiumRS, IntegrationRuntimeSsisCatalogPricingTierStandard}
}
```

Reply:
can you add a test with S0 or one of the new values se we can see if the SDK will actually support it? as i'm not sure it will

Reply:
I added the additional test cases, but they were skipped because of...
```
testcase.go:109: Acceptance tests skipped unless env 'TF_ACC' set
```